lib/client.js

```diff
@@ -301,23 +299,35 @@ class Client extends EventEmitter {
     return this.request('get', url, data, callback);
   }
 
-  getPendingAuth(url, data, callback, resolve) {
-    if (this.requestsPendingAuth) {
-      this.requestsPendingAuth.push({ url, data, callback, resolve });
-      return;
-    }
+  getPendingAuth(url, data, callback) {
+    return new Promise((resolve, reject) => {
+      if (this.requestsPendingAuth) {
+        this.requestsPendingAuth.push({ url, data, callback, resolve });
+        return;
+      }
 
-    this.requestsPendingAuth = [];
+      this.requestsPendingAuth = [];
 
-    this.authWithKey(() => {
-      this.get(url, data, (_err, result, headers) => {
-        if (callback) {
-          callback(null, result, headers);
-        }
-        resolve(result);
-      });
+      this.authWithKey(() => {
+        this.get(url, data, (err, result, headers) => {
+          if (typeof callback === 'function') {
+            callback(err, result, headers);
+          }
 
-      this.flushRequestsPendingAuth();
+          if (err) {
+            reject(err);
+            return;
+          }
+
+          resolve(result);
+        })
+        .then(() => {
+          this.flushRequestsPendingAuth();
+        })
+        .catch((err) => {
+          reject(err.message);
```

Comment:
Since a `promise` can only be `resolved` or `rejected` once during its life, it seems to me that this `catch` will not help us.

I think it would be better to wait for the promise and then do the rest like this:

```js
const result = await new Promise((resolve, reject) => { ... });

this.flushRequestsPendingAuth();

return result;
```

Reply:
Alternatively, moved the `.then( flushRequestsPendingAuth )` call to the outer promise.